app/validators/date_validator.rb

```diff
@@ -22,17 +20,13 @@ def date_of_birth_validations(record, attribute, value)
   def date_validations(record, attribute, value)
     return record.errors.add(attribute, :blank_date, article: article(attribute), attribute: humanize(attribute)) if options[:presence] && blank?(value)
 
-    return record.errors.add(attribute, :blank_date_fields, attribute: humanize(attribute), fields: blank_fields(value).to_sentence) if blank_fields(value).any?
+    return record.errors.add(attribute, :blank_date_fields, attribute: humanize(attribute), fields: blank_fields(value).to_sentence) if !blank?(value) && blank_fields(value).any?
```

Comment:
So we were accidentally marking missing date fields if nothing at all was submitted, which is basically the same as enforcing presence?

Reply:
Exactly.